### PR TITLE
Fix python log_level_overrides cannot be used on flink and other portable runners

### LINKED ISSUE #23632: [Bug]: Python sdk_harness_log_level_overrides option not compatible with runners Job Invoker
### What happened?

When running a Python Pipeline with Flink runner, FlinkJobInvoker tries to translate the pipeline options with a heuristic abc_def -> abcDef. For sdk_harness_log_level_overrides, this translates to sdkHarnessLogLevelOverrides which is an existing option in SdkHarnessOptions. But SdkHarnessOptions.sdkHarnessLogLevelOverrides is a Map<String, String>, while sdk_harness_log_level_overrides is a list. The following exception then thrown at job invoke:

```
INFO:apache_beam.utils.subprocess_server:SEVERE: Encountered Unexpected Exception for Preparation job_b8074b6e-d375-46e4-b449-b1419e74f048
INFO:apache_beam.utils.subprocess_server:java.lang.IllegalArgumentException: Error while populating display data for component 'org.apache.beam.sdk.options.ProxyInvocationHandler$PipelineOptionsDisplayData': Unable to parse representation
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3394)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.runners.core.construction.PipelineOptionsTranslation.toProto(PipelineOptionsTranslation.java:52)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.runners.flink.FlinkJobInvoker.createJobInvocation(FlinkJobInvoker.java:108)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.runners.flink.FlinkJobInvoker.invokeWithExecutor(FlinkJobInvoker.java:92)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.runners.jobsubmission.JobInvoker.invoke(JobInvoker.java:48)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.runners.jobsubmission.InMemoryJobService.run(InMemoryJobService.java:246)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.model.jobmanagement.v1.JobServiceGrpc$MethodHandlers.invoke(JobServiceGrpc.java:949)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:354)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.vendor.grpc.v1p48p1.io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
INFO:apache_beam.utils.subprocess_server:	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
INFO:apache_beam.utils.subprocess_server:	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
INFO:apache_beam.utils.subprocess_server:	at java.lang.Thread.run(Thread.java:750)
INFO:apache_beam.utils.subprocess_server:Caused by: com.fasterxml.jackson.databind.JsonMappingException: Error while populating display data for component 'org.apache.beam.sdk.options.ProxyInvocationHandler$PipelineOptionsDisplayData': Unable to parse representation
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._wrapAsIOE(DefaultSerializerProvider.java:509)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:482)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3389)
INFO:apache_beam.utils.subprocess_server:	... 18 more
INFO:apache_beam.utils.subprocess_server:Caused by: org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder$PopulateDisplayDataException: Error while populating display data for component 'org.apache.beam.sdk.options.ProxyInvocationHandler$PipelineOptionsDisplayData': Unable to parse representation
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder.include(DisplayData.java:789)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder.delegate(DisplayData.java:740)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:205)
INFO:apache_beam.utils.subprocess_server:	at com.sun.proxy.$Proxy15.populateDisplayData(Unknown Source)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder.include(DisplayData.java:780)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder.access$100(DisplayData.java:703)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData.from(DisplayData.java:81)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler$Serializer.serialize(ProxyInvocationHandler.java:774)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler$Serializer.serialize(ProxyInvocationHandler.java:721)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
INFO:apache_beam.utils.subprocess_server:	... 20 more
INFO:apache_beam.utils.subprocess_server:Caused by: java.lang.RuntimeException: Unable to parse representation
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler.getValueFromJson(ProxyInvocationHandler.java:586)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler.getValueFromJson(ProxyInvocationHandler.java:579)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler.access$700(ProxyInvocationHandler.java:95)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler$PipelineOptionsDisplayData.populateDisplayData(ProxyInvocationHandler.java:427)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.transforms.display.DisplayData$InternalBuilder.include(DisplayData.java:780)
INFO:apache_beam.utils.subprocess_server:	... 29 more
INFO:apache_beam.utils.subprocess_server:Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.LinkedHashMap<java.lang.String,java.lang.String>` from Array value (token `JsonToken.START_ARRAY`)
INFO:apache_beam.utils.subprocess_server: at [Source: UNKNOWN; byte offset: #UNKNOWN]
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1741)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1515)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromArray(StdDeserializer.java:222)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:447)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:32)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:424)
INFO:apache_beam.utils.subprocess_server:	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:32)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.PipelineOptionsFactory.deserializeNode(PipelineOptionsFactory.java:1807)
INFO:apache_beam.utils.subprocess_server:	at org.apache.beam.sdk.options.ProxyInvocationHandler.getValueFromJson(ProxyInvocationHandler.java:584)
INFO:apache_beam.utils.subprocess_server:	... 33 more
INFO:apache_beam.utils.subprocess_server:

```

This affects Python pipelines used `sdk_harness_log_level_overrides` option running on Flink, Spark, and Sazma runners, where these jobInvoker called PipelineOptionsTranslation.fromProto.

We should probably make the change from Python side, make sure that when the option passed to proto it is compatible with Java. i.e., having a well defined proto with map value.

### Issue Priority

Priority: 2

### Issue Component

Component: runner-flink

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #23632 which blocks #23631

* Use json.loads type for log_level_overrides argument

* Print error instead of fail the pipeline for unknown log level in Java

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py [hunk 1]
```diff
@@ -174,26 +185,23 @@ def test__set_log_level_overrides(self):
...
-      options_dict = {'sdk_harness_log_level_overrides': case}
-      sdk_worker_main._set_log_level_overrides(options_dict)
+      overrides = self._overrides_case_to_option_dict(case)
+      sdk_worker_main._set_log_level_overrides(overrides)
       for name, level in expected.items():
         self.assertEqual(logging.getLogger(name).getEffectiveLevel(), level)
 
   def test__set_log_level_overrides_error(self):
     test_cases = [
-        ({
```
Comment (Abacn) [inline sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py:184]: Now illegal json command as argument fails as early as pipeline option parsing. So that these test cases for error handling changed to valid json but unexpected arrays or invalid levels.

**Comment by Abacn** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py:184]: ack, thanks.

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 2]
```diff
@@ -133,6 +133,14 @@ def error(self, message):
     super().error(message)
 
 
+class _DictUnionAction(argparse.Action):
+  def __call__(self, parser, namespace, values, option_string=None):
+    if hasattr(namespace,
+               self.dest) and getattr(namespace, self.dest) is not None:
+      values.update(getattr(namespace, self.dest))
```
Comment (tvalentyn) [inline sdks/python/apache_beam/options/pipeline_options.py:140]: when a flag is specified multiple times, usually the last occurrence takes precedence. I think we should initialize or update self.dest with new values , not the opposite. WDYT?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java [hunk 3]
```diff
@@ -290,18 +303,24 @@ public SdkHarnessLogLevelOverrides addOverrideForName(String name, LogLevel logL
      * name}, or fully qualified Java {@link Package#getName() package name}, or custom logger name.
      * The {@code LogLevel} represents the log level and must be one of {@link LogLevel}.
      */
-    @JsonCreator
+    @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
```
Comment (tvalentyn) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java:306]: for my education, why was this necessary?

File: sdks/python/apache_beam/options/pipeline_options.py (hunk 2 above)
Comment (Abacn) [reply sdks/python/apache_beam/options/pipeline_options.py:140]: Thanks. This makes sense. Will do and document.

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 4]
```diff
@@ -134,11 +134,16 @@ def error(self, message):
 
 
 class _DictUnionAction(argparse.Action):
+  """
+  argparse Action take union of json loads values. If a key is specified in more
+  than one of the values, the last value takes precedence.
+  """
   def __call__(self, parser, namespace, values, option_string=None):
-    if hasattr(namespace,
```
Comment (tvalentyn) [inline sdks/python/apache_beam/options/pipeline_options.py]: Given dicts are mutable, I would prefer not to initialize a class member with a value passed into a function, even though it will probably be fine in this case. Consider the following instead:

```
    if not hasattr(namespace,
      self.dest) or getattr(namespace, self.dest) is None:
      setattr(namespace, self.dest, {})
    getattr(namespace, self.dest).update(values)
```

**Comment by tvalentyn** [comment]
seeing some test failures
```
>     getattr(namespace, self.dest).update(values)
E     ValueError: dictionary update sequence element #0 has length 7; 2 is required
```

**Comment by Abacn** [comment]
> seeing some test failures
> 
> ```
> >     getattr(namespace, self.dest).update(values)
> E     ValueError: dictionary update sequence element #0 has length 7; 2 is required
> ```

Thanks @tvalentyn . This is due to non-dict json loads fail early now. Test fixed. Ready for review again now.

**Comment by tvalentyn** [comment]
test failure is a known flake: https://github.com/apache/beam/issues/21104

**Review by tvalentyn — APPROVED** [review]
